container.go

```diff
@@ -422,29 +423,35 @@ func attachExistingIO(response *tasks.GetResponse, ioAttach cio.Attach) (cio.IO,
 
 // loadFifos loads the containers fifos
 func loadFifos(response *tasks.GetResponse) *cio.FIFOSet {
-	path := getFifoDir([]string{
+	fifos := []string{
 		response.Process.Stdin,
 		response.Process.Stdout,
 		response.Process.Stderr,
-	})
+	}
 	closer := func() error {
-		return os.RemoveAll(path)
+		var (
+			err  error
+			dirs []string
+		)
+		for _, fifo := range fifos {
+			if isFifo, _ := sys.IsFifo(fifo); isFifo {
+				err = os.Remove(fifo)
```

Comment:
maybe:
```go
if rerr = os.Remove(fifo); err == nil {
   err = rerr
}
```

Reply:
makes sense; also changed `dirs` to a map so we aren't wasting 3 iterations of `os.Remove()` on the same directory potentially.